RobotCode/src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -1,76 +1,62 @@
 package frc.robot.subsystems;
 
-// import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.revrobotics.SparkPIDController;
+import com.revrobotics.CANSparkBase.ControlType;
 import com.revrobotics.CANSparkBase.IdleMode;
 
 import frc.robot.constants.Constants;
 import frc.robot.constants.Ports;
 import frc.robot.constants.ShooterConstants;
-import edu.wpi.first.math.controller.PIDController;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import frc.lib.core.motors.TeamSparkMAX;
 
 public class ShooterSubsystem extends SubsystemBase
 {
 	// Creates objects
-	private double shooterMotorPower, feederMotorPower;
+	private double desiredShooterVelocity;
 
-	private final PIDController shooterPid, feederPid;
-	private final TeamSparkMAX ShooterMotorMain, ShooterMotorSub, FeederMotorMain, FeederMotorSub;
-
-	private static final double VOLTAGE = 12;
+	private SparkPIDController shooterPid;
+	private TeamSparkMAX shooterMotorMain, shooterMotorSub;
 
 	public ShooterSubsystem()
 	{
-		// Sets default shooter motor power to 0.25 and feeder to 0
-		shooterMotorPower = 0.25;
-		feederMotorPower = 0;
-
-		// Initializes the Pid
-		shooterPid = new PIDController(ShooterConstants.SHOOTER_KP, ShooterConstants.SHOOTER_KI,
-				ShooterConstants.SHOOTER_KD);
-		feederPid = new PIDController(ShooterConstants.FEEDER_KP, ShooterConstants.FEEDER_KI,
-				ShooterConstants.FEEDER_KD);
+		// Sets default shooter motor power
+		desiredShooterVelocity = ShooterConstants.SHOOTER_REST_VELOCITY;
 
 		// Initializes motor object
-		ShooterMotorMain = new TeamSparkMAX("Left Shooter Motor Main", Ports.SHOOTER_MOTOR_MAIN);
-		ShooterMotorSub = new TeamSparkMAX("Right Shooter Motor Main", Ports.SHOOTER_MOTOR_SUB);
+		shooterMotorMain = new TeamSparkMAX("Left Shooter Motor Main", Ports.SHOOTER_MOTOR_MAIN);
+		shooterMotorSub = new TeamSparkMAX("Right Shooter Motor Main", Ports.SHOOTER_MOTOR_SUB);
 
-		FeederMotorMain = new TeamSparkMAX("Left Shooter Motor Sub", Ports.FEEDER_MOTOR_MAIN);
-		FeederMotorSub = new TeamSparkMAX("Right Shooter Motor Sub", Ports.FEEDER_MOTOR_SUB);
+		shooterMotorSub.follow(shooterMotorMain, true);
 
-		// Inverts the right side
-		ShooterMotorSub.setInverted(true);
-		FeederMotorSub.setInverted(true);
+		shooterMotorMain.enableVoltageCompensation(Constants.MAX_VOLTAGE);
```

Comment:
Motors have voltage comp done inside their constructor now, so this is unnecessary unless you're giving a voltage other than 12.